test/segment-loader.test.js

```diff
@@ -20,6 +23,75 @@ const ogAddSegmentMetadataCue_ = SegmentLoader.prototype.addSegmentMetadataCue_;
 
 SegmentLoader.prototype.addSegmentMetadataCue_ = function() {};
 
+QUnit.module('SegmentLoader Isolated Functions');
+
+QUnit.test('illegalMediaSwitch detects illegal media switches', function(assert) {
```

Comment:
I think it would be good to adda test that passes `type === 'audio'` since the audio segment loader will also be calling this function